### DISCUSSION TITLE
Snowflake driver > adding parameter doesn't work

### DISCUSSION THREAD

**Discussion by JoMrx** [dbeaver/dbeaver#21658] [General]
I'm trying to add a parameter to the URL template, but after the change of the template, all connection attempts fail with response like "unknown host"
Default Template (works fine): jdbc:snowflake://{host}[:port]/?[db={database}]
Failing Template (fails): jdbc:snowflake://{host}[:port]/?allowUnderscoresInHost=true&[db={database}]
What's wrong there, or is there another way of adding a jdbc parameter? The default for the Underscores in Host have been changed on jdbc driver side, and newer driver versions need this parameter set.
Another point: No docu found for the spelling of the template; what does [ .. ] = the square brackets do here?

**Comment by LonwoLonwo** [comment]
Hello @JoMrx 

Could you please provide the error from the falling template using?

Also, you can add property this way:

![2023-10-27 13_34_55-Connect to a database](https://github.com/dbeaver/dbeaver/assets/45152336/4fe97617-7a8f-4fad-9c90-6fd62f6952d2)

> Another point: No docu found for the spelling of the template; what does [ .. ] = the square brackets do here?

Could you please provide me  a screen from the place? To understand - what exactly template is it?

**Comment by JoMrx** [comment]
Hello Anastasiya,

thanks a lot, the parameter set on the “driver properties” tab has worked. It simply wasn’t self-explaining, that “parameters” are the same as “properties”.

However, I have no “Test Connection” button in my dbeaver version (latest MacOS aarch64 version. That’s by intention?

The URL templarte is on the 1st tab, and I have tried to add the additional parameter there in the URL, but that only served errors ( “host not found” ). While trying, I found that I don’t understand the square brackets there, and as the additional parameter on the URL didn’t work, I was trying to understand the syntax of the template. It’s the default setting, so the square brackets were meant to be there, hence without documentation.

**Reply by LonwoLonwo** [reply]
Square brackets are used for necessary variables, like database. You can connect without the database name. But not - without the hostname.

> However, I have no “Test Connection” button in my dbeaver version (latest MacOS aarch64 version. That’s by intention?

Oh no. Probably it is better to redownload.
